js/components/LoginScreen.js

```diff
@@ -119,6 +118,17 @@ class LoginScreen extends React.Component {
     }).start()
   }
 
+  handleIpdSelection = (idp) => {
+    // if the selected idp is the demo one simulate a sucessfull login
+    if(idp.id == 'demo') {
```

Comment:
since this check comes un in a few places, I would create a utility method `isDemoIdp(idp: IdentityProvider): boolean`